### PR TITLE
Relocate endian to <bit>

### LINKED ISSUE #61: P1612R1 Relocating endian To <bit>
[P1612R1](https://wg21.link/P1612R1) Relocating `endian` To `<bit>`

### PR DESCRIPTION
**Pull request by AdamBucior**
# Description

Resolves #61 

# Checklist

Be sure you've read README.md and understand the scope of this repo.

If you're unsure about a box, leave it unchecked. A maintainer will help you.

- [x] Identifiers in product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
- [ ] The STL builds successfully and all tests have passed (must be manually
  verified by an STL maintainer before automated testing is enabled on GitHub,
  leave this unchecked for initial submission).
- [x] These changes introduce no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate
  or trivially copyable, etc.).
- [x] These changes were written from scratch using only this repository,
  the C++ Working Draft (including any cited standards), other WG21 papers
  (excluding reference implementations outside of proposed standard wording),
  and LWG issues as reference material. If they were derived from a project
  that's already listed in NOTICE.txt, that's fine, but please mention it.
  If they were derived from any other project (including Boost and libc++,
  which are not yet listed in NOTICE.txt), you *must* mention it here,
  so we can determine whether the license is compatible and what else needs
  to be done.

### CODE REVIEW

File: stl/inc/bit [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+// ENUM CLASS endian
```
Comment (CaseyCarter) [inline stl/inc/bit]: We've been discussing not adding SHOUTY BANNERS in new headers, preparatory to dropping them everywhere.
```suggestion
```

Comment (StephanTLavavej) [reply stl/inc/bit]: Filed as #306.

File: stl/inc/bit [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// bit standard header
```
Comment (StephanTLavavej) [inline stl/inc/bit]:
```suggestion
// bit standard header (core)
```

A subset of our headers, like `<type_traits>`, are usable without any runtime support machinery. We indicate this by saying `(core)` in their introductory comment. This is true for `<bit>` as well (even after the power-of-2 etc. functions are added, they'll be powered by compiler builtins, nothing separately compiled).

**Comment by barcharcraz** [comment]
my PR is up at #310  

The race is on ;D

**Comment by StephanTLavavej** [comment]
I am porting this to our Microsoft-internal MSVC repo and running tests; when that's successful, I'll merge your PR here, and we'll rebase Charlie's PR accordingly.

**Comment by StephanTLavavej** [comment]
Merged for VS 2019 16.5 Preview 2. Thanks again, and congratulations on implementing your first C++20 feature here!

**Pull request merged**